File: Windows Client/FacultyManager/FacultyManager/Model/Operations/ServerResponse/AccountResponse.cs
using System;

namespace FacultyManager.Model.Operations.ServerResponse
{
    public class AccountResponse : ServerOperation
    {       
        public string Email => _email;

        public string FirstName => _firstName;

        public string LastName => _lastName;

        public string IdNumber => _idNumber;

        public string PhoneNumber => _phoneNumber;
        public bool IsApproved => _isApproved;

        public string Birthday => _birthday;
        public int Permission => _permission;
        private string _email;
        private string _firstName;
        private string _lastName;
        private string _idNumber;
        private string _phoneNumber;
        private string _birthday;
        private int _permission;
        private bool _isApproved;
        public AccountResponse(short opCode) : base(opCode)
        {
            _email = "";
            _firstName = "";
            _lastName = "";
            _idNumber = "";
            _phoneNumber = "";
            _birthday = "";
            _permission = -1;
            _isApproved = false;
        }

        public override bool pushByte(byte nextByte)
        {
            if(nextByte == ';')
                return true;
            if(nextByte==(byte)0)
            {
                if(_email.Equals(""))
                    _email = bytesToString();
                else if(_firstName.Equals(""))
                    _firstName = bytesToString();
                else if(_lastName.Equals(""))
                    _lastName = bytesToString();
                else if(_idNumber.Equals(""))
                    _idNumber = bytesToString();
                else if(_phoneNumber.Equals(""))
                    _phoneNumber = bytesToString();
                else if(_birthday.Equals(""))
                    _birthday = bytesToString();
                else if (_permission == -1)
                    _permission = Int32.Parse(bytesToString());
                else
                    _isApproved = bytesToBoolean();
            }
            else
                pushNextByte(nextByte);
            return false;
        }
    }
}

File: Server/ServiceLayer/Objects/Operations/Server/AccountResponse.java
package ServiceLayer.Objects.Operations.Server;

import ServiceLayer.Objects.ServerOperation;
import ServiceLayer.Protocol;

public class AccountResponse extends ServerOperation {
    private String email;
    private String firstName;
    private String lastName;
    private String idNumber;
    private String phoneNumber;
    private String birthday;
    private boolean isApproved;

    public AccountResponse(short opCode, String email, String firstName, String lastName, String idNumber, String phoneNumber, String birthday, boolean isApproved) {
        super(opCode);
        this.email = email;
        this.firstName = firstName;
        this.lastName = lastName;
        this.idNumber = idNumber;
        this.phoneNumber = phoneNumber;
        this.birthday = birthday;
        this.isApproved = isApproved;
    }


    public byte[] encode() {
        byte[] bytes = new byte[email.length() + firstName.length() + lastName.length() + idNumber.length() + phoneNumber.length() + phoneNumber.length() + birthday.length() + 11];
        bytes[0] = (byte)((getOpCode() >> 8) & 0xFF);
        bytes[1] = (byte)(getOpCode() & 0xFF);
        int index = 2;
        index = AddStringToByteArray(email, bytes, index);
        bytes[index++] = 0;
        index = AddStringToByteArray(firstName, bytes, index);
        bytes[index++] = 0;
        index = AddStringToByteArray(lastName, bytes, index);
        bytes[index++] = 0;
        index = AddStringToByteArray(idNumber, bytes, index);
        bytes[index++] = 0;
        index = AddStringToByteArray(phoneNumber, bytes, index);
        bytes[index++] = 0;
        index = AddStringToByteArray(birthday, bytes, index);
        bytes[index++] = 0;
        bytes[index++] = isApproved?(byte)2:1;
        bytes[index++] = 0;
        bytes[index] = (byte)';';
        return bytes;
    }

    @Override
    public void execute(Protocol protocol) {

    }
}
